www-apps/gitea/gitea-1.7.4-r1.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit golang-vcs-snapshot systemd user
+
+EGO_PN="code.gitea.io/gitea"
+KEYWORDS="~amd64 ~arm"
+
+DESCRIPTION="A painless self-hosted Git service"
+HOMEPAGE="https://gitea.io"
+SRC_URI="https://github.com/go-gitea/gitea/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="pam sqlite"
+
+CDEPEND="pam? ( sys-libs/pam )"
+DEPEND="${CDEPEND}
+	dev-go/go-bindata"
+RDEPEND="${CDEPEND}
+	dev-vcs/git"
+
+DOCS=( custom/conf/app.ini.sample CONTRIBUTING.md README.md )
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+pkg_setup() {
+	enewgroup git
+	enewuser git -1 /bin/bash /var/lib/gitea git
+}
+
+gitea_make() {
+	local my_tags=(
+		bindata
+		$(usev pam)
+		$(usex sqlite 'sqlite sqlite_unlock_notify' '')
+	)
+	local my_makeopt=(
+		DRONE_TAG=${PV}
+		TAGS="${my_tags[@]}"
+	)
+	GOPATH=${WORKDIR}/${P}:$(get_golibdir_gopath) emake "${my_makeopt[@]}" "$1"
+}
+
+src_compile() {
+	gitea_make generate
+	gitea_make build
+}
+
+src_test() {
+	gitea_make test
+}
+
+src_install() {
+	einstalldocs
+	dobin gitea
+	newconfd "${FILESDIR}"/gitea.confd-r1 gitea
+	newinitd "${FILESDIR}"/gitea.initd-r2 gitea
+	systemd_dounit "${FILESDIR}"/gitea.service
+	diropts -m0750 -o git -g git
+	keepdir /etc/gitea /var/log/gitea /var/lib/gitea /var/lib/gitea/custom /var/lib/gitea/data
+}
+
+pkg_postinst() {
+	ewarn "The default configuration path changed to /etc/gitea/app.ini"
+	ewarn "Please migrate your configuration and then you can remove the"
+	ewarn "previously used /var/lib/gitea/conf directory."
+	if [[ ! -e "${EROOT}/etc/gitea/app.ini" ]]; then
+		elog "No app.ini found, copying initial config over"
+		cp "${FILESDIR}"/app.ini "${EROOT}"/etc/gitea/ || die
```

Comment:
According to https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#d-ed-root-eroot-no-longer-have-a-trailing-slash there is no trailing slash in eapi 7.

Reply:
I don't see anything wrong using an empty configuration. It will bootstrap you and the rest is your configuration or configuration management. We can endlessly modify the sample configuration file in the ebuild, but it's really a personal choice. If you want, you just delete the empty configuration (that actually allows you use gitea right away) and use the sample from the docs. Depending on your USE flag (sqlite/-sqlite), during the installation you'll see sqlite as the database choice. The sample configuration file is now in DOCS so in /usr/share/docs/gitea...

Reply:
> The app.ini of filesdir is basically useless and misleading.
> 
>     * no RUN_MODE (default: dev)
>
If a fresh installation is started, the RUN_MODE is set to prod
>     * no HTTP_ADDR (default: 0.0.0.0  -> listing on all interfaces)
I believe you should have a firewall in place. What is the reason to install a git web if not used over the net?
>     * compiled with sql useflag but sql not enable
After USE=sqlite emerge gitea I see sqlite as the database type.

Reply:
Yes, fine just differ form the approach to use the app.ini.sample as inital config and override it (like ssh does).
But your way might be better, since gitea still changes a lot....